src/render/View3D.cpp

```diff
@@ -498,6 +514,42 @@ void View3D::paintGL()
     glCheckError();
 }
 
+
+QImage View3D::renderScreenShot(int width, int height)
+{
+    if (m_badOpenGL)
+        return QImage();
+
+    double dPR = getDevicePixelRatio();
+    if (width < 0)
+        width = this->width();
+
+    if (height < 0)
+        height = this->height();
+
+    // Note: the reason for the following two lines and the subsequent downsampling of the
+    //       resulting image is to make the radius of rendered points consistent between the
+    //       gui and screenshot on displas with dPR != 1.
+    width *= dPR;
+    height *= dPR;
+
+    QOpenGLFramebufferObject framebuffer(QSize(width, height), QOpenGLFramebufferObject::Depth);
```

Comment:
This class is Qt5 only, so we'd be breaking with Qt4 compatibility by using this.  I guess let's get it all working, and then I'll need to decide.

(TBH, I'm kinda keen to give up on Qt4 because the number of different OpenGL and OS versions is already enough to be a real drag on limited development resources...)